plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+import MagicIcon from '../../assets/images/icons/magic.svg';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+const DESCRIPTION_MAX_LENGTH = 300;
+
+export function WriteItForMeButtonContainer() {
+	const [ fetching, setFetching ] = useState( false );
+	const generatingContentPhraseInterval = useRef< number >();
+	const titleEl = useRef< HTMLInputElement >(
+		document.querySelector( '#title' )
+	);
+	const [ productTitle, setProductTitle ] = useState< string >(
+		titleEl.current?.value || ''
+	);
+	const tinyEditor = useTinyEditor();
+
+	useEffect( () => {
+		titleEl.current?.addEventListener( 'keyup', ( e ) =>
+			setProductTitle( ( e.target as HTMLInputElement ).value || '' )
+		);
```

Comment:
Do we need to perform cleanup here as well when the component is unmounted by removing the EventListener?
```
  return () => {
    titleEl.current?.removeEventListener('keyup', keyupHandler);
  };
  ```

Reply:
Good call 👍🏻